conductor/src/main.rs

```diff
@@ -66,9 +70,24 @@ fn main() {
                     .expect("Could not start UI servers!");
             }
 
-            // TODO wait for a SIGKILL or SIGINT instead here.
-            loop {
-                sleep(Duration::from_secs(1))
+            for _sig in termination_signals.forever() {
+                let mut conductor_guard = CONDUCTOR.lock().unwrap();
+                let conductor = std::mem::replace(&mut *conductor_guard, None);
+                let refs = Arc::strong_count(&CONDUCTOR);
+                if refs == 1 {
+                    println!("Gracefully shutting down conductor...");
+                } else {
```

Comment:
what's the difference between 'graceful' shutdown and 'explicit' shutdown?

(context, I've lost the ability to fully track/understand some of the more sophisticated Rust code that's being written by y'all now)

Reply:
I guess the idea is that if the reference count for the `CONDUCTOR` Arc is 1 there is nobody else (no other thread) alive and running for this conductor and we don't need to explicitly call `conductor.shutdown()` as that is called in the `Drop` implementation.

But I don't completely get the difference either since we take ownership of the conductor in line 75 with `mem::replace`, right? So that object will be dropped at the end of the for-body's scope either way.. Isn't that right, @maackle?

Reply:
Yes, this bit is a little confusing, but I figured it's better to have something than nothing. I was thinking that other threads would still have access to a conductor in the explicit shutdown case, but @lucksus is right that the conductor gets dropped either way. However, if there are other threads referencing CONDUCTOR at this point, they would wind up with a `None` when they expected `Some(Conductor)`, so this warning at least explains why. I could modify the warning message a little, but I think it still mostly makes sense like this.